homeworks/vitali.timolyanov_XX-NF-XX/homework_5/task_3_inner_copy.js

```diff
@@ -0,0 +1,21 @@
+function innerCopy(object, depth = 1) {
+    const copy = Object.assign({}, object);
+    if (depth > 0) {
+        const objectEntries = Object.entries(copy).filter(entry => typeof entry[1] === 'object');
+        objectEntries.forEach((entry) => {
+            copy[entry[0]] = innerCopy(entry[1], depth - 1);
+        });
+    }
+    return copy;
+}
```

Comment:
If property has an array, than `Object.assign()` will convert that array into new object with keys as indexes e.g. `Object.assign( {}, ['a'] )` returns `{ 0: 'a' }`
It is possible to skip arrays with `&& !Array.isArray(entry[1])` added to the end of the 4th line.
As arrays wasn't mentioned in the task I left it with this behavior. 
There are many different ways of handling array. At this point it can be converted to an object, copied to a new array or just left as the reference to the same array. And what if this array has other arrays or objects inside? What if there was something like this `new Number(9)`? Should it be treated as a number or just left as an object?

Reply:
no worries, I just was wondering :)

`typeof null` also returns `'object'`, but I guess you are aware of it

anyway, deep cloning objects is a quite complex operation out of scope of this task :)